docs/dockerfile_to_build_custom_logstash_image.md

```diff
@@ -0,0 +1,72 @@
+# Dockerfile to build a custom Logstash image
+
+To build an image that is not available in the [official Docker repository tags](https://hub.docker.com/r/opensearchproject/logstash-oss-with-opensearch-output-plugin/tags), or to add specific plugins to your image, you can create a custom Dockerfile.
+
+The process varies depending on whether you want to build an image with `Logstash 8.x` versions or `Logstash 7.x` versions
+
+## Logstash 8.x
+
+Create this `Dockerfile` to build an image with `Opensearch 2.0.2` for **`Logstash 8.x`**
+
+``` Dockerfile
+ARG APP_VERSION
+
+FROM docker.elastic.co/logstash/logstash-oss:${APP_VERSION}
+RUN logstash-plugin install --version 7.1.1 logstash-integration-aws
+RUN logstash-plugin install --version 2.0.2 logstash-output-opensearch
+```
+
+## Logstash 7.x
+
+For **`Logstash 7.x`** , the `logstash output opensearch gem` needs to be build.
+
+### Build logstash output opensearch plugin gem
+
+1. Clone `logstash-output-opensearch repo` 
+
+    ```sh
+    git clone https://github.com/opensearch-project/logstash-output-opensearch.git
+    ```
+
+2. Checkout the tag for the plugin version you want to build. For the version [2.0.2](https://github.com/opensearch-project/logstash-output-opensearch/tree/2.0.2) for example
+
+    ```sh
+    git checkout 2.0.2
+    ```
+
+
+3. Remove [this line that adds the json version spec](https://github.com/opensearch-project/logstash-output-opensearch/blob/2.0.2/logstash-output-opensearch.gemspec#L49) 
```

Comment:
Btw, why is this needed? Let's explain here.

Reply:
Added a little explanation.